### PR TITLE
add observationUpdate

### PR DESCRIPTION
**Pull request by karissa**

### CODE REVIEW

File: api.js [hunk 1]
```diff
@@ -88,13 +82,43 @@ Api.prototype.observationCreate = function (req, res, m) {
...
+    obses = flatObs(m.id, obses)
 
+    body(req, function (err, obs) {
+      if (err) {
+        res.statusCode = 400
+        res.end('couldnt parse body json: ' + err.toString())
+        return
+      }
+      var old = obses[0]
+      var newObs = Object.assign(old, obs)
```
[the remark this answers is not in the document]
Comment (karissa) [reply api.js]: yeah I think it's nice in an HTTP API to not have to fetch the item and pass around all of it's values in order to update it, so you could have a view that knows about the item's ID and the new values and update those without worrying about losing the underlying data.

I wonder if it's worth it to sanitize the data so only approved fields will be patched?

File: api.js [hunk 2]
```diff
@@ -88,13 +82,43 @@ Api.prototype.observationCreate = function (req, res, m) {
...
+    }
+    obses = flatObs(m.id, obses)
 
+    body(req, function (err, obs) {
+      if (err) {
+        res.statusCode = 400
+        res.end('couldnt parse body json: ' + err.toString())
+        return
+      }
+      var old = obses[0]
```
[the remark this answers is not in the document]
Comment (karissa) [reply api.js]: OK cool, and `opts` in this case is the observation object?

[the remark this answers is not in the document]
Comment (noffle) [reply api.js]: It's another param to [osm-p2p-db#put](https://github.com/digidem/osm-p2p-db#osmputid-doc-opts-cb)

[the remark this answers is not in the document]
Comment (karissa) [reply api.js]: ok I attempted it, although it feels like this ought to be handled on a lower level someday

**Pull request merged**